product_origin/models/product_template.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+
+    _inherit = 'product.template'
+
+    country_id = fields.Many2one(
```

Comment:
This should not be modifiable otherwise you get discrepancies with the state.
IMO should be calculated/readonly